tests/testthat/test_stan_betareg.R

```diff
@@ -23,78 +23,60 @@ library(betareg)
-  dat <- data.frame(dat$y, dat$x)
-  colnames(dat) <- c("y", "x")
-  expect_error(fit <- stan_betareg(y ~ x, link = "logit", seed = SEED, sparse = TRUE,
-                                   prior = NULL, prior_intercept = NULL,
-                                   data = dat, algorithm = "optimizing"))
+  dat <- simple_betareg_data(200)
+  expect_error(
+    stan_betareg(y ~ x, link = "logit", seed = SEED, sparse = TRUE, data = dat),
+    "unknown arguments: sparse"
+  )
 })
 
 # test QR 
-context("stan_betareg QR error")
-test_that("QR errors when x and/or z predictors are <= 1", {
-  dat <- list()
-  dat$N <- 200
-  dat$x <- rnorm(dat$N, 2, 1)
-  dat$z <- rep(0, dat$N)
-  dat$mu <- binomial(link="logit")$linkinv(1+0.2*dat$x)
-  dat$phi <- 20
-  dat$y <- rbeta(dat$N, dat$mu * dat$phi, (1 - dat$mu) * dat$phi)
-  dat <- data.frame(dat$y, dat$x, dat$z)
-  colnames(dat) <- c("y", "x", "z")
-  expect_error(fit <- stan_betareg(y ~ x, link = "logit", seed = SEED, QR = TRUE,
-                           prior = NULL, prior_intercept = NULL,
-                           data = dat, algorithm = "optimizing"))
-  expect_error(fit <- stan_betareg(y ~ x | z, link = "logit", seed = SEED, QR = TRUE,
-                                   prior = NULL, prior_intercept = NULL,
-                                   data = dat, algorithm = "optimizing"))
+test_that("QR errors when number of x and/or z predictors is <= 1", {
+  dat <- simple_betareg_data(200, draw_z = TRUE)
+  expect_error(
+    stan_betareg(y ~ x, link = "logit", seed = SEED, QR = TRUE, data = dat),
+    "'QR' can only be specified when there are multiple predictors"
+  )
+  expect_error(
+    stan_betareg(y ~ x | z, link = "logit", seed = SEED, QR = TRUE, data = dat),
+    "'QR' can only be specified when there are multiple predictors"
+  )
 })
 
-context("stan_betareg QR tests")
-test_that("QR works when x and/or z predictors are >= 1", {
-  dat <- list()
-  dat$N <- 200
-  dat$x <- rnorm(dat$N, 2, 1)
-  dat$z <- rep(0, dat$N)
-  dat$mu <- binomial(link="logit")$linkinv(1+0.2*dat$x)
-  dat$phi <- 20
-  dat$y <- rbeta(dat$N, dat$mu * dat$phi, (1 - dat$mu) * dat$phi)
-  dat <- data.frame(dat$y, dat$x, dat$z)
-  colnames(dat) <- c("y", "x", "z")
-  fit1 <- stan_betareg(y ~ x + z, link = "logit", seed = SEED, QR = TRUE,
+test_that("QR works when number of x and/or z predictors is >= 1", {
+  dat <- simple_betareg_data(200, draw_z = TRUE)
+  SW(fit1 <- stan_betareg(y ~ x + z, link = "logit", seed = SEED, QR = TRUE,
                                    prior = NULL, prior_intercept = NULL,
-                                   data = dat, algorithm = "optimizing")
+                                   data = dat, algorithm = "optimizing"))
+  expect_stanreg(fit1)
+  
   fit2 <- stan_betareg(y ~ x | z, link = "logit", seed = SEED, QR = TRUE,
                                    prior = NULL, prior_intercept = NULL,
                                    data = dat, algorithm = "optimizing")
+  expect_stanreg(fit2)
```

Comment:
Yeah the test should pass but I wrote it wrong. QR decomposition is not possible if the user  doesn't specify more than one x predictor or z predictor.